geode-core/src/main/java/org/apache/geode/internal/cache/tier/sockets/ClientProtocolHandshaker.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more contributor license
+ * agreements. See the NOTICE file distributed with this work for additional information regarding
+ * copyright ownership. The ASF licenses this file to You under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the License. You may obtain a
+ * copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package org.apache.geode.internal.cache.tier.sockets;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+
+import org.apache.geode.cache.IncompatibleVersionException;
+import org.apache.geode.security.server.Authenticator;
+
+public interface ClientProtocolHandshaker {
+  public Authenticator handshake(InputStream inputStream, OutputStream outputStream)
+      throws IOException, IncompatibleVersionException;
+
+  public boolean shaken();
```

Comment:
We need some javadoc on here explaining what this method is supposed to do. Shaken does not automatically make me think that this is `completed` or even `successful`

Reply:
changed the name, updated.